test/e2e/util/util.go

```diff
@@ -1397,6 +1403,57 @@ func RunQueryAtTime(ctx context.Context, log logr.Logger, prometheusClient prome
 	}, nil
 }
 
+// getExportedMetrics exec curl command in HO pod metrics endpoint and return metric values if any
+func getExportedMetrics(ctx context.Context, log logr.Logger, config *restclient.Config) (map[string]*dto.MetricFamily, error) {
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+	hoPods, err := kubeClient.CoreV1().Pods("hypershift").List(context.Background(),
+		metav1.ListOptions{LabelSelector: "app=operator,hypershift.openshift.io/operator-component=operator"})
+	if err != nil {
+		return nil, fmt.Errorf("couldn't retrieve HO Pods: %v", err)
+	}
+	if len(hoPods.Items) == 0 {
+		return nil, fmt.Errorf("coldn't find any HO Pods")
+	}
+	hoPod := hoPods.Items[0]
+
+	podIP := hoPod.Status.PodIP
+	curlArg := "http://" + podIP + ":9000/metrics"
+	req := kubeClient.CoreV1().RESTClient().Post().Resource("pods").Name(hoPod.Name).Namespace("hypershift").SubResource("exec").VersionedParams(
```

Comment:
we already have a util to exec into pods here: https://github.com/openshift/hypershift/blob/main/test/e2e/util/pod_exec.go
also see: https://github.com/openshift/hypershift/blob/main/test/e2e/util/util.go#L1143 for usage example

Reply:
That's great. Thank you!